### ISSUE TITLE
About training code and some question about the DDPO

### ISSUE THREAD

**Issue by Yxxxb** [RLHF-V/RLHF-V#10]
Thanks for your amazing work!

I have several questions:
1. I do not find the training code and the shell, I wonder if they are not released.
2. The DPO algorithm is to label the output of the pre-trained model with artificial prefer and disprefer. But RLHF-V is a manual correction of the results of the model output. Is that mean the output of the model is all disprefered, and the modified ones are all prefered?

**Comment by Yxxxb** [comment]
Can you provide the code for this part, if possible.

**Comment by Haoye17** [comment]
Hi @Yxxxb ! Thanks for your interest in our work! Here are the answers to your questions, hope they can help!

1. We provide our training code and scripts in [Muffin](https://github.com/thunlp/Muffin) repo and we provide an instruction to train our model in readme, you can follow the instructions in the [install](https://github.com/RLHF-V/RLHF-V?tab=readme-ov-file#install) and [RLHF-V training](https://github.com/RLHF-V/RLHF-V?tab=readme-ov-file#rlhf-v-training) section to train the model.

2. It is correct that the outputs of the model are all dispreferred, and the modified ones are all preferred in our data. When doing annotation, if the model output does not contain hallucinations, we will not use it to construct a correctional pair. Therefore, the remaining model outputs all contain hallucinations and are dispreferred.
